src/ctim/schemas/weakness.cljc

```diff
@@ -0,0 +1,218 @@
+             :description (str "how effective the detection method may be in "
+                               "detecting the associated weakness"))
+    (f/entry :effectiveness_notes c/MedString
+             :description (str "provides additional discussion of the "
+                               "strengths and shortcomings of this detection "
+                               "method"))))
+  :reference "[DetectionMethodsType](https://cwe.mitre.org/documents/schema/#DetectionMethodsType)")
+
+(def-map-type Mitigation
+  (concat
+   (f/required-entries
+    (f/entry :description c/MedString
+             :description (str "a description of this individual mitigation "
+                               "including any strengths and shortcomings of "
+                               "this mitigation for the weakness")))
+   (f/optional-entries
+    (f/entry :phases [v/SoftwarePhase]
+             :description (str "indicates the development life cycle phase "
+                               "during which this particular mitigation may "
+                               "be applied"))
+    (f/entry :strategy v/MitigationStrategy
+             :description (str "a general strategy for protecting a system "
+                               "to which this mitigation contributes"))
+    (f/entry :effectiveness v/Effectiveness
+             :description (str "summarizes how effective the mitigation may be "
+                               "in preventing the weakness"))
+    (f/entry :effectiveness_notes c/MedString)))
+  :reference "[PotentialMitigationsType](https://cwe.mitre.org/documents/schema/#PotentialMitigationsType)")
+
+(def-entity-type Weakness
+  {:description weakness-desc
+   :reference weakness-desc-link}
+  c/base-entity-entries
+  c/describable-entity-entries
+  c/sourcable-object-entries
+  (f/required-entries
+   (f/entry :type WeaknessTypeIdentifier
+            :description (str "The fixed value " type-identifier))
+   (f/entry :description c/Markdown
+            :description (str "should be short and limited to the key points "
+                              "that define this weakness")))
+  (f/optional-entries
+   (f/entry :structure v/WeaknessStructure
+            :description (str "defines the structural nature of the weakness"))
+   (f/entry :languages [Language]
+            :description "Applicable Languages")
+   (f/entry :os [OperatingSystem]
+            :description "Applicable operating systems")
+   (f/entry :architectures [Architecture]
+            :description "Applicable architectures")
+   (f/entry :paradigms [Paradigm]
+            :description "Applicable paradigms")
+   (f/entry :technologies [Technology]
+            :description "Applicable technologies")
+   (f/entry :background_details c/Markdown
+            :description (str "information that is relevant but not related to "
+                              "the nature of the weakness itself"))
+   (f/entry :alternate_terms [c/ShortString]
```

Comment:
Alternate terms can have optional descriptions:

```
{:alternate-term
   {:term "Tapjacking",
    :description
    "\"Tapjacking\" is similar to clickjacking, except it is used for mobile applications in which the user \"taps\" the application instead of performing a mouse click."}}
```